### ISSUE TITLE
Invalid argument(s): Failed to load dynamic library (dlopen failed: library "libabc.so"     not found)

### ISSUE THREAD

**Issue by xuchaoqian** [flutter/flutter#58968]
It's fine to run my app on a real phone, but failed to run it on an emulator.

**Expected results:** 
It should load the dynamic library correctly.

**Actual results:** 
Invalid argument(s): Failed to load dynamic library (dlopen failed: library "libabc.so"     not found)

<details>
  <summary>Logs</summary>

```
    The following ArgumentError was thrown attaching to the render tree:
    Invalid argument(s): Failed to load dynamic library (dlopen failed: library "libabc.so"
    not found)
    When the exception was thrown, this was the stack:
    #0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:13:55)
    #1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22:12)
    #2      new FFI._ (package:abc/src/ffi.dart:64:26)
    #3      new FFI (package:abc/src/ffi.dart:57:23)
    #4      new Runtime (package:abc/src/runtime.dart:22:17)
    #5      _MyAppState.initState (package:abc_example/main.dart:25:21)
    #6      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4640:58)
    #7      ComponentElement.mount (package:flutter/src/widgets/framework.dart:4476:5)
I/flutter: #8      Element.inflateWidget (package:flutter/src/widgets/framework.dart:3446:14)
    #9      Element.updateChild (package:flutter/src/widgets/framework.dart:3214:18)
    #10     RenderObjectToWidgetElement._rebuild (package:flutter/src/widgets/binding.dart:1148:16)
    #11     RenderObjectToWidgetElement.mount (package:flutter/src/widgets/binding.dart:1119:5)
    #12     RenderObjectToWidgetAdapter.attachToRenderTree.<anonymous closure> (package:flutter/src/widgets/binding.dart:1061:17)
    #13     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2607:19)
    #14     RenderObjectToWidgetAdapter.attachToRenderTree (package:flutter/src/widgets/binding.dart:1060:13)
    #15     WidgetsBinding.attachRootWidget (package:flutter/src/widgets/binding.dart:941:7)
    #16     WidgetsBinding.scheduleAttachRootWidget.<anonymous closure> (package:flutter/src/widgets/binding.dart:922:7)
    (elided 11 frames from class _RawReceivePortImpl, class _Timer, dart:async, and dart:async-patch)
```

```
Analyzing abc...                                       
No issues found! (ran in 2.8s)
```

```
[✓] Flutter (Channel stable, v1.17.3, on Mac OS X 10.15.5 19F101, locale en)
    • Flutter version 1.17.3 at /Users/abc/flutter
    • Framework revision b041144f83 (4 days ago), 2020-06-04 09:26:11 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/abc/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/abc/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.8.4

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.11.0

[✓] Proxy Configuration
    • HTTP_PROXY is set
    • NO_PROXY is localhost,127.0.0.0,127.0.0.1
    • NO_PROXY contains 127.0.0.1
    • NO_PROXY contains localhost

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)
```

</details>

**Comment by dcharkes** [comment]
Can you provide a minimal repro?

How are you linking the library? dynamically or statically for the real app?

**Comment by xuchaoqian** [comment]
@dcharkes Thanks for your fast response.

I have created a repo: https://github.com/xuchaoqian/flutter_plugin_abc.git

I used dynamically linking for the real phone and the emulator.

**Comment by xuchaoqian** [comment]
@dcharkes 

I have found the cause of the problem.

I set abiFilters in build.gradle as following: 

        externalNativeBuild {
            cmake {
                abiFilters 'arm64-v8a', "armeabi-v7a", "x86_64"
            }
        }

But my emulator is 32bit. They don't match. After switching to 64bit emulator, everything works fine.

Thank you for your awesome works on dart ffi.  It enpowers the dart/flutter community to use the the high-performace and rich c/c++ libraries.
